src/components/TextArea.tsx

```diff
@@ -0,0 +1,121 @@
+import { forwardRef, useImperativeHandle, useRef, useState } from 'react';
+import {
+  TextInput,
+  StyleSheet,
+  type StyleProp,
+  type ViewStyle,
+  type TextInputProps,
+  type TextInputFocusEventData,
+  type NativeSyntheticEvent,
+} from 'react-native';
+
+import InputContainer from './InputContainer';
+
+import useTheme from '../hooks/useTheme';
+
+import { convertHexToRgba } from '../utils/uiUtils';
+
+type Props = TextInputProps & {
+  style?: StyleProp<ViewStyle>;
+  error?: boolean;
+  label?: string;
+  hint?: string;
+  numberOfLines?: number;
+  maxLength?: number;
+  onChangeText?: (text: string) => void;
+};
+
+export type InputRef = { focus: () => void; blur: () => void };
+
+export const TextArea = forwardRef<InputRef, Props>(
```

Comment:
Can we add a character counter to the bottom right of the text area (inside of the outline)? We did this on glowy and worked great! You can even copy it from there